tools/tap-driver

```diff
@@ -212,8 +212,11 @@ class TapDriver(Driver):
             return
         description = description.lstrip()
 
-        # Special case if description starts with this, then skip
-        if description.lower().startswith("# skip"):
+        # Parse out a directive from description, if any
+        (description, unused, directive) = description.partition("#")
+
+        # Special case if directive starts with this, then skip
+        if directive.lstrip().lower().startswith("skip"):
```

Comment:
This looks wrong. This function starts out with a whole line like

    ok 23 # skip known to not work here

which the original code turned into `description == "# skip known to not work here"`, and pass that on to `result_skip()`. But the new `partition()` call above now turns this into `description == ""` (which is passed on), and `directive == "skip known to not work here"` (which is thrown away).

I'm not sure what this is actually supposed to fix. Can you please be more verbose in the commit log? Perhaps you want to split this as

    (_, _, description) = description.partition("#")

? The visible effect of that would be to drop the `#` from the reported result.

Reply:
> which the original code turned into description == "# skip known to not work here"

The problem is that the original code doesn't work like that. After [line 207](https://github.com/cockpit-project/cockpit/pull/9924/files/17e5ead8defe8c145f381acae051b68901e2f68a..821fa7707ac4e4c0029d2d7d870115a7e1530c03#diff-b5a45ed5cac3c210f90850e7d32e0116L207), description becomes: `23 # skip known to not work here`. This second `partition()` is to scrub "23 # ".

Reply:
Indeed this is really hard to follow - this should *really* use a regular expression, that's so much easier to write and read, and more robust. Anyway, this *is* broken. Run `make check TESTS=test-locale` on master, you will see
```
SKIP: test-locale 7 # skip -- locale not available
PASS: test-locale 8 /locale/set-language/de-de
invalid or unusable locale: __xx;%%%.UTF-8
SKIP: test-locale 9 # skip -- locale not available
```
with this PR, the description gets killed, as I wrote above:
```
SKIP: test-locale 7 
PASS: test-locale 8 /locale/set-language/de-de
invalid or unusable locale: __xx;%%%.UTF-8
SKIP: test-locale 9 
```